pkg/api/mock/matcher/matcher_test.go

```diff
@@ -0,0 +1,187 @@
+	)
+
+	BeforeEach(func() {
+		a = mock.NewMockAPI()
+
+		objects = make([]*testObject, 0, 8)
+
+		for i := 0; i < 4; i++ {
+			obj0 := testObject{TestFieldA: "text 0"}
+			obj1 := testObject{TestFieldA: "text 1"}
+			a.FakeExisting(&obj0)
+			a.FakeExisting(&obj1)
+			objects = append(objects, &obj0, &obj1)
+		}
+	})
+
+	Context("ContainElementNTimes", func() {
+		It("can test if the API list contains elements n times", func() {
+			Expect(a.Existing()).To(ContainElementNTimes(Object(&testObject{}), 8))
+			Expect(a.Existing()).To(ContainElementNTimes(Object(&testObject2{}), 0))
+		})
+
+		It("returns error when ACTUAL not of type array/slice/map", func() {
+			matcher := ContainElementNTimes(nil, 5)
+			_, err := matcher.Match(true)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(HavePrefix("ContainElementNTimes matcher expects an array/slice/map."))
+		})
+
+		It("fallbacks to equal matcher when element is no types.GomegaMatcher", func() {
+			matcher := ContainElementNTimes("test", 2)
+			Expect(matcher.Match([]string{"test", "toast", "test"})).To(BeTrue())
+			Expect(matcher.Match([]string{"test", "toast"})).To(BeFalse())
+		})
+
+		It("returns error when sub-match returns error", func() {
+			matcher := ContainElementNTimes(ContainElement(1), 2)
+			_, err := matcher.Match([]string{"test1", "test2"})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(HavePrefix("ContainElement matcher expects an array/slice/map."))
+		})
+
+		It("has (Negated)FailureMessage", func() {
+			Expect(ContainElementNTimes(nil, 0).FailureMessage(nil)).To(ContainSubstring("to contain n elements matching"))
+			Expect(ContainElementNTimes(nil, 0).NegatedFailureMessage(nil)).To(ContainSubstring("not to contain n elements matching"))
+		})
+	})
+
+	Context("CRUD Matcher", func() {
+		It("can match objects created n times", func() {
+			Expect(ContainElement(Created(5)).Match(a.All())).To(BeFalse())
+			Expect(ContainElement(Created(0)).Match(a.All())).To(BeTrue())
+
+			err := a.Create(context.TODO(), &testObject{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ContainElementNTimes(Created(1), 1).Match(a.All())).To(BeTrue())
+		})
+
+		It("can match objects updated n times", func() {
+			Expect(ContainElement(Updated(5)).Match(a.All())).To(BeFalse())
+			Expect(ContainElement(Updated(0)).Match(a.All())).To(BeTrue())
+
+			obj := testObject{}
+			a.FakeExisting(&obj)
+			err := a.Update(context.TODO(), &obj)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ContainElementNTimes(Updated(1), 1).Match(a.All())).To(BeTrue())
+		})
+
+		It("can match objects destroyed n times", func() {
+			Expect(ContainElement(Destroyed(5)).Match(a.All())).To(BeFalse())
+			Expect(ContainElement(Destroyed(0)).Match(a.All())).To(BeTrue())
+
+			obj := testObject{}
+			a.FakeExisting(&obj)
+			err := a.Destroy(context.TODO(), &obj)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ContainElementNTimes(Destroyed(1), 1).Match(a.All())).To(BeTrue())
```

Comment:
can you add a test case for `Destroyed(2)`? Forgot why we have that as number, in which case we destroy the same Object multiple times.. ^^'

Reply:
CloudDNS Zones have their name as identifier and can potentially be created & destroyed multiple times.